### PR TITLE
CLN/TST: Convert `tests/tseries/offset` to pytest idioms

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

* Converted `setup_method` to `pytest.fixture`s where needed
* Removed `Base` class (tests were moved in #49017) and refactored to a stand alone function

### CODE REVIEW

File: pandas/tests/tseries/offsets/conftest.py [hunk 1]
```diff
@@ -29,3 +32,11 @@ def month_classes(request):
     Fixture for month based datetime offsets available for a time series.
     """
     return request.param
+
+
+@pytest.fixture
```
Comment (jreback) [inline pandas/tests/tseries/offsets/conftest.py:37]: how about date or dt for the name?

File: pandas/tests/tseries/offsets/test_offsets.py [hunk 2]
```diff
@@ -121,17 +156,19 @@ class TestCommon(Base):
         "Nano": Timestamp("2011-01-01T09:00:00.000000001"),
     }
 
+
+class TestCommon:
     def test_immutable(self, offset_types):
         # GH#21341 check that __setattr__ raises
-        offset = self._get_offset(offset_types)
+        offset = _create_offset(offset_types)
```
Comment (jreback) [inline pandas/tests/tseries/offsets/test_offsets.py:163]: shouldn't you make a *new* fixture that does this

**Review by jreback — CHANGES_REQUESTED** [review]
a couple of comments

File: pandas/tests/tseries/offsets/test_offsets.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/tseries/offsets/test_offsets.py:163]: This function is also used in the testing class helper methods so I think this is better (for now) as a separate function

File: pandas/tests/tseries/offsets/conftest.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/tseries/offsets/conftest.py:37]: Sure changed to `dt`

**Comment by mroeschke** [comment]
Addressed and green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**